fdbclient/include/fdbclient/Tenant.h

```diff
@@ -58,18 +55,9 @@ struct TenantMapEntry {
 
 	template <class Ar>
 	void serialize(Ar& ar) {
-		KeyRef subspace;
-		if (ar.isDeserializing) {
-			serializer(ar, id, subspace);
-			if (id >= 0) {
-				initPrefix(subspace);
-			}
-		} else {
-			ASSERT(prefix.size() >= 8 || (prefix.empty() && id == -1));
-			if (!prefix.empty()) {
-				subspace = prefix.substr(0, prefix.size() - 8);
-			}
-			serializer(ar, id, subspace);
+		serializer(ar, id);
+		if (ar.isDeserializing && id >= 0) {
+			prefix = idToPrefix(id);
 		}
```

Comment:
```suggestion
		if constexpr (ar.isDeserializing) {
			if (id >= 0) {
				prefix = idToPrefix(id);
			}
		}
```
Nit, but depending on how often we do this, the above will be slightly more efficient since it will not do the comparison if we're serializing